docs/reference/template-tags.md

```diff
@@ -0,0 +1,191 @@
+
+### Usage
+
+```django
+{% area context, areaName with contextOptions %}
+```
+
+**Example:**
+```django
+{% area data.page, 'main' with {
+  '@apostrophecms/image': {
+    sizes: '(min-width: 600px) 45vw, (min-width: 1140px) 530px'
+  }
+} %}
+```
+
+### Arguments
+
+#### `context`
+
+The area's document context, either a page (`data.page`), piece (`data.piece`), or widget (`data.widget`). The area field must be defined in the field schema for that context. See the [template data](/guide/template-data.md) section for more on each `data` property.
+
+#### `areaName`
+
+The name (a string) of the area field as defined in the field schema.
+
+#### `contextOptions` (optional)
+
+The context options object is added after area tag arguments following the `with` keyword. It is an object with keys matching the names of widget types allowed in that particular area field. Each key is assigned a value that will be passed into the widget template as `data.contextOptions`. See the [image widget guide](/guide/core-widgets.md#image-widget) for an example.
+
+Context options are optional for all core and official Apostrophe widget types.
+
+## component
+
+The `component` tag inserts an asynchronous component into the template. See the [async components guide](/guide/async-components.md) for more on using this feature.
+
+### Usage
+
+```django
+{% component 'moduleName:componentName' with data %}
+```
+
+**Example:**
+```django
+{% component 'product:newest' with { max: 3 } %}
+```
+
+### Arguments
+
+#### `moduleName:componentName`
+
+The primary argument is a combination of the name of a module name and the name of an async component from that module, separated by a colon. All async components belong to a specific module, though multiple modules may have components with the same name.
+
+#### `data` (optional)
+
+The data argument, following the `with` keyword, is available in the async component template as `data`. It can be any data type, however it is a best practice to make it an object with subproperties so using the values are clearer in templates.
+
+## fragment
+
+The `fragment` tag *declares* a template fragment that will be inserted elsewhere with [`render`](#render) or [`rendercall`](#rendercall). See the [template fragments guide](/guide/fragments.md) for more on using this feature.
+
+This tag must be closed with an `endfragment` tag.
+
+### Usage
+
+```django
+{% fragment name(arguments) %}
+  {# Fragment markup #}
+{% endfragment %}
+```
+
+**Examples:**
+```django
+{% fragment button(text, action, options = {}) %}
+  <button class="o-button {{ options.class }}" data-action="{{ action }}">
+    {{ text }}
+  </button>
+{% endfragment %}
+```
+
+```django
+{% fragment card(data = {}) %}
+  <section class="o-card">
+    <h2>{{ data.heading }}</h2>
+    <div>
+      {{ rendercaller() }}
+    </div>
+  </section>
+{% endfragment %}
+```
+
+### Arguments
+
+#### `name()`
```

Comment:
What's with the parens?

Reply:
OK I see, it's because the whole declaration is an argument to the fragment tag. I think it would be clearer to me if "params..." appeared inside the parens or something.